pkg/controllers/restore-controller/handler.go

```diff
@@ -0,0 +1,213 @@
+	if IsInitStatus(rst) {
+		rst.Status = apis.RSTCStorStatusInProgress
+		_, err := c.clientset.OpenebsV1alpha1().CStorRestores(rst.Namespace).Update(rst)
+		if err != nil {
+			klog.Errorf("Failed to update restore:%s status : %v", rst.Name, err.Error())
+			return "", err
+		}
+
+		err = volumereplica.CreateVolumeRestore(rst)
+		if err != nil {
+			klog.Errorf("restore creation failure: %v", err.Error())
+			return string(apis.RSTCStorStatusFailed), err
+		}
+		c.recorder.Event(rst, corev1.EventTypeNormal, string(common.SuccessCreated), string(common.MessageResourceCreated))
+		klog.Infof("restore creation successful: %v, %v", rst.ObjectMeta.Name, string(rst.GetUID()))
+		return string(apis.RSTCStorStatusDone), nil
+	} else if IsPendingStatus(rst) {
+		klog.Infof("Updating restore:%s status to %v", rst.Name, apis.RSTCStorStatusInit)
+		return string(apis.RSTCStorStatusInit), nil
+	}
+	return "", nil
+}
+
+// getCStorRestoreResource returns a restore object corresponding to the resource key
+func (c *RestoreController) getCStorRestoreResource(key string) (*apis.CStorRestore, error) {
+	// Convert the key(namespace/name) string into a distinct name
+	klog.V(1).Infof("Finding restore for key:%s", key)
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil, nil
+	}
+
+	rst, err := c.clientset.OpenebsV1alpha1().CStorRestores(ns).Get(name, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Restore resource for key:%s is missing", name)
+		return nil, err
+	}
+	return rst, nil
+}
+
+// IsPendingStatus is to check if the restore is in a pending state.
+func IsPendingStatus(rst *apis.CStorRestore) bool {
+	if string(rst.Status) == string(apis.RSTCStorStatusPending) {
+		return true
+	}
+	return false
+}
+
+// IsInProgressStatus is to check if the restore is in in-progress state.
+func IsInProgressStatus(rst *apis.CStorRestore) bool {
+	if string(rst.Status) == string(apis.RSTCStorStatusInProgress) {
+		return true
+	}
+	return false
+}
+
+// IsInitStatus is to check if the restore is in init state.
+func IsInitStatus(rst *apis.CStorRestore) bool {
+	if string(rst.Status) == string(apis.RSTCStorStatusInit) {
+		return true
+	}
+	return false
+}
+
+// IsDoneStatus is to check if the restore is completed or not
+func IsDoneStatus(rst *apis.CStorRestore) bool {
+	if string(rst.Status) == string(apis.RSTCStorStatusDone) {
+		return true
+	}
+	return false
+}
+
+// IsFailedStatus is to check if the restore is failed or not
+func IsFailedStatus(rst *apis.CStorRestore) bool {
+	if string(rst.Status) == string(apis.RSTCStorStatusFailed) {
+		return true
+	}
+	return false
+}
+
+// IsRightCStorPoolMgmt is to check if the restore request is for particular pod/application.
+func IsRightCStorPoolMgmt(rst *apis.CStorRestore) bool {
+	if os.Getenv(string(common.OpenEBSIOCStorID)) == rst.ObjectMeta.Labels["cstorpool.openebs.io/uid"] {
```

Comment:
key needs to be updated `"cstorpool.openebs.io/uid"` -->  `types.CStorPoolInstanceUIDLabelKey`